pages/search.js

```diff
@@ -143,292 +185,165 @@ const NoResults = () => (
-      }
-      return {
-        error,
-        results: [],
-        nextURL: null,
-        testNamesKeyed,
-        testNames,
-        countries
-      }
-    }
-
-    const measurements = msmtR.data
-    // drop results with probe_asn === 'AS0'
-    const results = measurements.results.filter(item => item.probe_asn !== 'AS0')
-
-    return {
-      error: null,
-      results,
-      nextURL: measurements.metadata.next_url,
-      testNamesKeyed,
-      testNames,
-      countries,
-    }
-  }
+const Search = ({testNames, testNamesKeyed, countries, query: queryProp }) => {
+  const router = useRouter()
+  const { query, replace, isReady } = router
 
-  constructor(props) {
-    super(props)
-    this.state = {
-      testNameFilter: props.router.query.test_name,
-      domainFilter: props.router.query.domain,
-      inputFilter: props.router.query.input,
-      categoryFilter: props.router.query.category_code,
-      countryFilter: props.router.query.probe_cc,
-      asnFilter: props.router.query.probe_asn,
-      sinceFilter: props.router.query.since,
-      untilFilter: props.router.query.until,
-      onlyFilter: props.router.query.only || 'all',
-      hideFailed: !props.router.query.failure,
-      results: props.results,
-      nextURL: props.nextURL,
-      error: props.error,
-
-      loading: false
-    }
-    this.getFilterQuery = this.getFilterQuery.bind(this)
-    this.onApplyFilter = this.onApplyFilter.bind(this)
-    this.loadMore = this.loadMore.bind(this)
-  }
+  const [nextURL, setNextURL] = useState(null)
+  const [loading, setLoading] = useState(false)
+  const [results, setResults] = useState([])
+  const [filters, setFilters] = useState({})
+  const [error, setError] = useState(null)
 
-  componentDidMount () {
-    const { query, replace } = this.props.router
+  useEffect(() => {
+    const query = query || queryProp
     const href = {
       pathname: '/search',
-      query
+      query: query
     }
     replace(href, href, { shallow: true })
-  }
 
-  shouldComponentUpdate (nextProps, nextState) {
-    if (this.props != nextProps) {
-      return true
-    }
-    if (this.state.results != nextState.results) {
-      return true
-    }
-    if (this.state.loading != nextState.loading) {
-      return true
-    }
-    return false
-  }
+    setLoading(true)
+    getMeasurements(query)
+      .then((res) => {
+        const results = res.data.results.filter(item => item.probe_asn !== 'AS0')
+        setLoading(false)
+        setResults(results)
+        setNextURL(res.data.metadata.next_url)
+      })
+      .catch((err) => {
+        console.error(err)
+        const error = serializeError(err)
+        setLoading(false)
+        setError(error)
+      })
+  }, [])
 
-  loadMore() {
-    axios.get(this.state.nextURL)
+  const loadMore = () => {
+    axios.get(nextURL)
       .then((res) => {
-        // XXX update the query
         const nextPageResults = res.data.results.filter(item => item.probe_asn !== 'AS0')
```

Comment:
Same comment as above and I now realise this was already here, but maybe it's an opportunity to do some cleanup of this too.